endpoint-deprecation/README.md

```diff
@@ -0,0 +1,112 @@
+# Observability for Feature Deprecation
+
+Want to deprecate an API? Use [Pixie](https://github.com/pixie-io/pixie) to quickly determine:
+
+- Is this endpoint used?
+- Who is using this endpoint?
+
+[Can I deprecate this endpoint?]( https://blog.px.dev/endpoint-deprecation) is the accompanying blog post for this demo.
+
+## Prerequisites
+
+- If you don't already have one, set up a [Kubernetes cluster](https://docs.px.dev/installing-pixie/setting-up-k8s/).
+- [Install Pixie](https://docs.px.dev/installing-pixie/install-guides/) to your Kubernetes cluster.
+- Install the [Pixie CLI](https://docs.px.dev/installing-pixie/install-schemes/cli/#1.-install-the-pixie-cli) if you didn't install it in order to deploy Pixie in the step above.
+
+## Test Application
+
+1. Install [hey](https://github.com/rakyll/hey). Homebrew users can use:
+
+```
+brew install hey
+```
+
+2. Deploy an echo server to receive arbitrary client requests:
+
+```
+# echo-service will be created in the `default` namespace.
+kubectl apply -f demo-app.yaml
+```
+
+3. Get the external IP for `echo-service` and save it in an environment variable:
+
+```
+kubectl get services
+export ECHO_SERVICE_IP=<EXTERNAL IP>
+```
+
+4. Run the test load. `-H` is used to pass a custom HTTP header.
+
+```
+for i in {1..15}; do
+hey -H "Referer:https://example.com/" -H "API-KEY:abcdef12345" -n 550 "http://${ECHO_SERVICE_IP}/v1/catalog/"
+hey -H "Referer:https://px.dev/" -H "API-KEY:lkjlsdfsdfs" -n 50 "http://${ECHO_SERVICE_IP}/v1/catalog/$(uuidgen)/details"
+hey -H "Referer:https://example.com/" -H "API-KEY:abcdef12345" -n 50 "http://${ECHO_SERVICE_IP}/v1/catalog/$(uuidgen)/details"
+hey -H "Referer:https://google.com/" -H "API-KEY:sdfsdfsdfsd" -n 50 "http://${ECHO_SERVICE_IP}/v1/catalog/$(uuidgen)/details"
+hey -H "Referer:https://google.com/" -H "API-KEY:sdfsdfsdfsd" -n 50 "http://${ECHO_SERVICE_IP}/v2/catalog/$(uuidgen)"
+done
+```
+
+## Service Traffic Clustered by Logical Endpoint
+
+Copy the `service_endpoints_summary` folder locally, then run:
+
+```
+px live -f service_endpoints_summary -- -start_time '-30m' -service 'default'
```

Comment:
default/{servicename}? we probably want our docs to refelct the correct data structures for service name which is {ns}/{service} even when a freeform filter is allowed